app/scripts/controllers/monitoring.js

```diff
@@ -0,0 +1,311 @@
+      };
+      $scope.logCanRun.pods[pod.metadata.name] = !(_.includes(['New', 'Pending', 'Unknown'], pod.status.phase));
+    };
+
+    var setDeploymentLogVars = function(deployment) {   
+      $scope.logOptions.deployments[deployment.metadata.name] = {
+        container: $filter("annotation")(deployment, "pod")
+      };
+      var deploymentVersion = $filter("annotation")(deployment, "deploymentVersion");
+      if (deploymentVersion) {
+        $scope.logOptions.deployments[deployment.metadata.name].version = deploymentVersion;
+      }        
+      $scope.logCanRun.deployments[deployment.metadata.name] = !(_.includes(['New', 'Pending'], $filter('deploymentStatus')(deployment)));
+    };
+
+    var setBuildLogVars = function(build) {
+      $scope.logOptions.builds[build.metadata.name] = {
+        container: $filter("annotation")(build, "buildPod")
+      };
+      $scope.logCanRun.builds[build.metadata.name] = !(_.includes(['New', 'Pending', 'Error'], build.status.phase));
+    };
+
+    var filterPods = function() {
+      ageFilteredPods = _.filter($scope.pods, function(pod) {
+        if (!$scope.filters.hideOlderResources) {
+          return true;
+        }
+        if (pod.status.phase !== 'Succeeded' && pod.status.phase !== 'Failed') {
+          return true;
+        }
+        return false;
+      });
+      $scope.filteredPods = filterForKeyword(ageFilteredPods);
+    };
+
+    var isIncompleteBuild = $filter('isIncompleteBuild');
+    var buildConfigForBuild = $filter('buildConfigForBuild');
+    var filterBuilds = function() {
+      var fiveMinutesAgo = moment().subtract(5, 'm');
+      ageFilteredBuilds = _.filter($scope.builds, function(build) {
+        if (!$scope.filters.hideOlderResources) {
+          return true;
+        }        
+        if (isIncompleteBuild(build)) {
+          return true;
+        }
+        var buildConfigName = buildConfigForBuild(build);
+        if (buildConfigName) {
+          return $scope.latestBuildByConfig[buildConfigName].metadata.name === build.metadata.name;
+        }
+        
+        // Otherwise this is a one-off build, keep any of those that finished in the last 5 minutes, if we
+        // don't have a completionTimestamp for some reason then fallback to creationTimestamp
+        var completed = moment(build.status.completionTimestamp || build.metadata.creationTimestamp);
+        return completed.isAfter(fiveMinutesAgo);        
+      });
+      $scope.filteredBuilds = filterForKeyword(ageFilteredBuilds);
+    };
+
+    var deploymentStatus = $filter('deploymentStatus');
+    var filterDeployments = function() {
+      ageFilteredDeployments = _.filter($scope.deployments, function(deployment) {
+        if (!$scope.filters.hideOlderResources) {
+          return true;
+        }        
+        var status = deploymentStatus(deployment);
+        if (status !== 'Complete' && status !== 'Failed') {
+          return true;
+        }
+        return false;
```

Comment:
not quite, because "Active" is not considered in progress but we still want to show that, i can switch it to be

```
if (deploymentIsInProgress(deployment) || status === 'Active') {
```